packages/x-date-pickers/src/internals/components/PickersModalDialog.tsx

```diff
@@ -94,6 +94,7 @@ export function PickersModalDialog(props: React.PropsWithChildren<PickersModalDi
     <Dialog
       open={open}
       onClose={onDismiss}
+      {...slotProps?.dialog}
       {...componentsProps?.dialog}
```

Comment:
We can just replace `componentsProps` by `slotProps`
All the pickers are doing the fallback, so the internal components don't have to handle both.

And I think it's the same for the lines after (cc @alexfauquette)

If you use

```
export default function BasicDatePicker() {
    return (
        <LocalizationProvider dateAdapter={AdapterDayjs}>
            <DemoContainer components={['DatePicker']}>
                <MobileDatePicker label="Date Picker" componentsProps={{ dialog: { fullScreen: true } }} />
            </DemoContainer>
        </LocalizationProvider>
    );
}
```

You can see that it still receives `slotProps` inside `PickersModalDialog`

Reply:
In summary:

1. The picker (e.g: `DesktopDatePicker`) does the fallback the is `useXXXDefaultizedProps`

```tsx
  const slots = themeProps.slots ?? uncapitalizeObjectKeys(themeProps.components);
  return {
    // ... other props
    slots: { toolbar: DatePickerToolbar, ...slots },
    slotProps: themeProps.slotProps ?? themeProps.componentsProps,
  }
```

2. The headless responsive picker (e.g: `useDesktopPicker`) only receives the new `slots` / `slotProps` API and passes it to the layout, dialog, tooltip, etc...

```tsx
```

3. The internal components only consumes the new API

Reply:
As long as it's not exported from a public endpoint of the library, for me it's not a breaking change on the public API.

Reply:
I confirm, the 3 components (Date, Time, and DateTime) do the merging in [`shared.ts`](https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/DatePicker/shared.tsx) so Dialog should get `slots`, and `componentsProps` is not provided anymore.


If at the same time you can remove it in props typing, it would be a plus 👍

Reply:
I've gone through all the internal components and [removed `components` and `componentsProps`](https://github.com/mui/mui-x/pull/8509/commits/a8e017f025b2034c5e4ad27f3686af67092e9041).
The `TimeClock` component needed some adjustment as it is public and needs to work with both `components` and `slots` API. 😉